src/roles/netdata/tasks/main.yml

```diff
@@ -0,0 +1,49 @@
+---
+
+## Add test to see if it's already installed and skip if it is
+- name: Check if Netdata is present
+  stat:
+    path: /opt/netdata
+  register: netdata_present
+
+- debug: msg="netdata not installed"
+  when: netdata_present.stat.exists == False
+
+- debug: msg="netdata installed"
+  when: netdata_present.stat.exists == True
+
+## It will be running once installed
+- name: Install Intel/AMD 64bit static build of Netdata
+  shell: bash <(curl -Ss https://my-netdata.io/kickstart-static64.sh) --dont-wait
+  args:
+    executable: /bin/bash
+  become: no
+  when: netdata_present.stat.exists == False
+
+- name: Ensure port 20000 allowed for list of servers
+  include_role:
+    name: firewall_port
+  vars:
+    firewall_action: open
+    firewall_port: 20000
```

Comment:
Port 20000 should be opened to the world on load balancers. As such it should be in `src/roles/haproxy/tasks/main.yml` as another port to open in the task `Ensure port 80 and 443 open for haproxy`. Really that task should also include a conditional for if the load balancer is handling external connections. Since port 20000 should only be opened if `m_install_netdata` is true you'll need to append to a list of ports as shown below with `fully_open_ports`. I haven't tested this but it looks like it should work.

```yaml
- name: Create list of ports to open to the world
  set_fact:
    fully_open_ports:
      - 80
      - 443

- name: Add port 20000 to world-accessible list if installing netdata
  set_fact:
    fully_open_ports: "{{ fully_open_ports }} + [ 20000 ]"
  when: m_install_netdata

- name: Ensure port {{ fully_open_ports | join(', ') }} open to the world for haproxy
  include_role:
    name: firewall_port
  vars:
    firewall_action: open
    firewall_port: "{{ item }}"
    firewall_protocol: tcp
    firewall_zone: "{{m_private_networking_zone|default('public')}}"
  with_items: "{{ fully_open_ports }}"
  when:
    - load_balancer_handle_external # <------ Add conditional for external load balancer only
    - (docker_skip_tasks is not defined or not docker_skip_tasks)
    - >
      ('load-balancers' in groups and inventory_hostname in groups['load-balancers'])
      or
      ('load-balancers-meza-external' in groups and inventory_hostname in groups['load-balancers-meza-external'])
```